src/components/WeeklySummary/WeeklySummary.jsx

```diff
@@ -119,7 +134,30 @@ export class WeeklySummary extends Component {
       .startOf('isoWeek')
       .add(5, 'days');
 
+    const uploadDate =
+      summary === ''
+        ? dueDate
+        : weeklySummaries && weeklySummaries[0] && weeklySummaries[0].uploadDate;
+    const uploadDateLastWeek =
+      summaryLastWeek === ''
+        ? dueDateLastWeek
+        : weeklySummaries && weeklySummaries[1] && weeklySummaries[1].uploadDate;
+    const uploadDateBeforeLast =
+      summaryBeforeLast === ''
+        ? dueDateBeforeLast
+        : weeklySummaries && weeklySummaries[2] && weeklySummaries[2].uploadDate;
+    const uploadDateThreeWeeksAgo =
+      summaryThreeWeeksAgo === ''
+        ? dueDateThreeWeeksAgo
+        : weeklySummaries && weeklySummaries[3] && weeklySummaries[3].uploadDate;
```

Comment:
```suggestion
    const uploadDateXWeeksAgo = (x) => {
      summary === ''
        ? dueDate
        : weeklySummaries && weeklySummaries[x] && weeklySummaries[x].uploadDate;
    }
    const uploadDate = uploadDateXWeeksAgo(0);
    const uploadDateLastWeek = uploadDateXWeeksAgo(1);
    const uploadDateBeforeLast = uploadDateXWeeksAgo(2);
    const uploadDateThreeWeeksAgo = uploadDateXWeeksAgo(3);
```
Reducing redundancy.